tf_encrypted/keras/layers/normalization.py

```diff
@@ -112,46 +113,36 @@ def __init__(self,
     self.moving_variance_initializer = initializers.get(
         moving_variance_initializer)
 
-    if beta_regularizer:
-      raise NotImplementedError(
-          arg_not_impl_msg.format("beta_regularizer", "BatchNormalization"),
-      )
-    if gamma_regularizer:
-      raise NotImplementedError(
-          arg_not_impl_msg.format("gamma_regularizer", "BatchNormalization"),
-      )
-    if beta_constraint:
-      raise NotImplementedError(
-          arg_not_impl_msg.format("beta_constraint", "BatchNormalization"),
-      )
-    if gamma_constraint:
-      raise NotImplementedError(
-          arg_not_impl_msg.format("gamma_constraint", "BatchNormalization"),
-      )
-    if renorm:
-      raise NotImplementedError(
-          arg_not_impl_msg.format("renorm", "BatchNormalization"),
-      )
-    if renorm_clipping:
-      raise NotImplementedError(
-          arg_not_impl_msg.format("renorm_clipping", "BatchNormalization"),
-      )
-    if fused:
-      raise NotImplementedError(
-          arg_not_impl_msg.format("fused", "BatchNormalization"),
-      )
-    if trainable:
-      raise NotImplementedError(
-          arg_not_impl_msg.format("trainable", "BatchNormalization"),
-      )
-    if virtual_batch_size:
-      raise NotImplementedError(
-          arg_not_impl_msg.format("virtual_batch_size", "BatchNormalization"),
-      )
-    if adjustment:
-      raise NotImplementedError(
-          arg_not_impl_msg.format("adjustment", "BatchNormalization"),
-      )
+    default_args_check(beta_regularizer,
+                       "beta_regularizer",
+                       "BatchNormalization")
+    default_args_check(gamma_regularizer,
+                       "gamma_regularizer",
+                       "BatchNormalization")
+    default_args_check(beta_constraint,
+                       "beta_constraint",
+                       "BatchNormalization")
+    default_args_check(gamma_constraint,
+                       "gamma_constraint",
+                       "BatchNormalization")
+    default_args_check(renorm,
+                       "renorm",
+                       "BatchNormalization")
+    default_args_check(renorm_clipping,
+                       "renorm_clipping",
+                       "BatchNormalization")
+    default_args_check(fused,
+                       "fused",
+                       "BatchNormalization")
+    default_args_check(trainable,
```

Comment:
I don't think we should be doing this for trainable -- we don't support training so this kwarg is meaningless for the tfe layer, so we should support either value

Reply:
Agreed, i will remove it.